### PR TITLE
PFPKI intune support

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Intune scep support.

# Impacts
pfpki

# Delete branch after merge
YES

# Checklist
- [x] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries
## New Features
* Microsoft Intune SCEP support

### CODE REVIEW

File: html/pfappserver/root/src/views/Configuration/clouds/_components/FormTypeAzure.vue [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      :column-label="$i18n.t('The client ID of the Azure service')"
+      :text="$i18n.t('Define the client ID defined in the Azure admin portal.')"
+    />
+
+    <form-group-client-secret namespace="client_secret"
+      :column-label="$i18n.t('The client secret of the Azure service')"
+      :text="$i18n.t('Define the client secret defined in the Azure admin portal.')"
+    />
+
+-->
```
Comment (satkunas) [inline html/pfappserver/root/src/views/Configuration/clouds/_components/FormTypeAzure.vue]: Remove

File: html/pfappserver/root/src/views/Configuration/_components/CloudsList.vue [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      <template v-slot:pageHeader>
+        <b-card-header>
+          <h4 class="mb-0" v-t="'Cloud Service'"></h4>
+        </b-card-header>
+      </template>
+      <template v-slot:buttonAdd>
+        <b-dropdown :text="$t('New Cloud')" variant="outline-primary">
+<!--          <b-dropdown-item :to="{ name: 'newCloud', params: { cloudType: 'Azure' } }">Microsoft Azure</b-dropdown-item>
+          <b-dropdown-item :to="{ name: 'newCloud', params: { cloudType: 'Google' } }">Google Cloud</b-dropdown-item>
+-->
```
Comment (satkunas) [inline html/pfappserver/root/src/views/Configuration/_components/CloudsList.vue]: Remove

File: html/pfappserver/root/src/views/Configuration/clouds/_components/FormTypeGoogle.vue [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    <form-group-client-id namespace="client_id"
+      :column-label="$i18n.t('The client ID of the Google service')"
+      :text="$i18n.t('Define the client ID defined in the Azure admin portal.')"
+    />
+
+    <form-group-client-secret namespace="client_secret"
+      :column-label="$i18n.t('The client secret of the Google service')"
+      :text="$i18n.t('Define the client secret defined in the Azure admin portal.')"
+    />
+-->
```
Comment (satkunas) [inline html/pfappserver/root/src/views/Configuration/clouds/_components/FormTypeGoogle.vue]: Remove

File: html/pfappserver/root/src/views/Configuration/clouds/_components/FormTypeGoogle.vue [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  FormGroupTenantId,
+  FormGroupClientId,
+  FormGroupClientSecret,
+}
+
+import { useForm as setup, useFormProps as props } from '../_composables/useForm'
+
+// @vue/component
+export default {
+  name: 'form-type-intune',
```
Comment (satkunas) [inline html/pfappserver/root/src/views/Configuration/clouds/_components/FormTypeGoogle.vue]: form-type-google

File: html/pfappserver/root/src/views/Configuration/clouds/_components/FormTypeAzure.vue [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  FormGroupTenantId,
+  FormGroupClientId,
+  FormGroupClientSecret,
+}
+
+import { useForm as setup, useFormProps as props } from '../_composables/useForm'
+
+// @vue/component
+export default {
+  name: 'form-type-intune',
```
Comment (satkunas) [inline html/pfappserver/root/src/views/Configuration/clouds/_components/FormTypeAzure.vue]: form-type-azure

**Review by satkunas — APPROVED** [review]

**Comment by satkunas** [comment]
@fdurand looks good. Fix the minor issue comments above. 

I will need to rework changes with #6397., which will cause breaking changes to this VueJS source. Please let me know once this merged.

File: go/caddy/pfpki/certutils/certutils.go [hunk 6]
```diff
@@ -356,3 +357,12 @@ func GetDNFromCert(namespace pkix.Name) map[string]string {
...
 	return attributeMap
 }
+
+func ThumbprintSHA1(cert *x509.Certificate) string {
+	sum := sha1.Sum(cert.Raw)
+	hex := make([]string, len(sum))
+	for i, b := range sum {
+		hex[i] = fmt.Sprintf("%02X", b)
+	}
+	return strings.Join(hex, ":")
```
Comment (jrouzierinverse) [inline go/caddy/pfpki/certutils/certutils.go:367]: If case does not matter
```
import "encoding/hex"
```

```golang
func ThumbprintSHA1(cert *x509.Certificate) string {
   return hex.EncodeToString(sha1.Sum(cert.Raw))
}
```

Comment (fdurand) [reply go/caddy/pfpki/certutils/certutils.go:367]: i want something like aa:bb:cc and not aabbcc

Comment (jrouzierinverse) [reply go/caddy/pfpki/certutils/certutils.go:367]: Then never mind

**Comment by fdurand** [comment]
BUMP

**Comment by fdurand** [comment]
@jrouzierinverse ready to merge ?

File: db/pf-schema-10.3.0.sql [hunk 7]
```diff
@@ -1500,6 +1500,8 @@ CREATE TABLE `pki_profiles` (
   `scep_enabled` int(11) DEFAULT NULL,
   `scep_challenge_password` varchar(255) DEFAULT NULL,
   `scep_days_before_renewal` varchar(255) DEFAULT NULL,
+  `cloud_enabled` int(11) DEFAULT NULL,
+  `cloud_service` varchar(255) DEFAULT NULL,
```
Comment (jrouzierinverse) [inline db/pf-schema-10.3.0.sql]: Remove from here and add it to here ```db/pf-schema-X.Y.sql```.

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

File: db/upgrade-10.0.0-10.1.0.sql [hunk 8]
```diff
@@ -58,6 +58,11 @@ SET STATEMENT sql_mode='NO_AUTO_VALUE_ON_ZERO' FOR
 \! echo "PIDs should be unique in the password table"
 CREATE UNIQUE INDEX IF NOT EXISTS `pid_password_unique` ON password(pid);
 
+\! echo "altering pki_profiles"
+ALTER TABLE pki_profiles
+    ADD COLUMN IF NOT EXISTS `cloud_enabled` int(11) AFTER scep_days_before_renewal,
+    ADD COLUMN IF NOT EXISTS `cloud_service` varchar(255) AFTER cloud_enabled;
+
```
Comment (jrouzierinverse) [inline db/upgrade-10.0.0-10.1.0.sql]: Remove from here and add it to ``` db/upgrade-X.X-X.Y.sql ```

File: debian/packetfence.conffiles [hunk 9]
```diff
@@ -5,6 +5,7 @@
 /usr/local/pf/conf/billing_tiers.conf
 /usr/local/pf/conf/chi.conf
 /usr/local/pf/conf/cluster.conf
+/usr/local/pf/conf/cloud.conf
```
Comment (jrouzierinverse) [inline debian/packetfence.conffiles:8]: Add conf/cloud.conf to ```.gitignore```

**Comment by fdurand** [comment]
Change done

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

File: db/upgrade-10.0.0-10.1.0.sql (hunk 8 above)
Comment (fdurand) [reply db/upgrade-10.0.0-10.1.0.sql]: it's already done in the previous commit.

**Pull request merged**